client/client.go

```diff
@@ -175,19 +175,7 @@ func (c *Client) downloadRoot() error {
 	var s *data.Signed
 	var raw []byte
 	if download {
-		logrus.Debug("downloading new root")
-		raw, err = c.remote.GetMeta(role, size)
-		if err != nil {
-			return err
-		}
-		hash := sha256.Sum256(raw)
-		if expectedSha256 != nil && !bytes.Equal(hash[:], expectedSha256) {
```

Comment:
You are correct, however I'd like gotuf to behave correctly and not rely on users of the library to implement additional protections/logic related to tuf. I'm looking to move more of the notary client functionality in to the gotuf client. At the moment we have a very split brained situation which makes testing harder and errors like this more likely.

At some point in the near future (probably after the 1.9 release), we're going to unify gotuf and notary within the notary repo as it has made less and less sense to keep them separate. At that point it will make even more sense to unify all the TUF logic into this gotuf client code, while the notary client will continue to handle things like setting up and managing keys, keeping and applying changelists, and other higher level operations.